server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -186,6 +186,7 @@
     @Override
     public void doClose() {
         engineSearcher.close();
+        this.searcher.clearQueryCancellations();
```

Comment:
Since we use the `ContextIndexSearcher` only in `DefaultSearchContext` I wonder if that would be cleaner to extend `Closeable` and to close the searcher here ?

Reply:
++, fixed in 50043b16a3a5c8ceed7777e4c4ec3f9d68b2add4